services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,271 @@
+from flask import Flask, Response
+from flask.ext.cors import CORS
+
+from dictionary_parser import Parser as DictionaryParser
+from report_parser import Parser, logger
+from timer import ResourceTimer
+
+# Token for GitHub authorization. Set on startup.
+GITHUB_AUTH_TOKEN = None
+
+DIR_NAME = 'dqa_repo'
+REMOTE_URL = 'github.com/PEDSnet/Data-Quality.git'
+
+
+class DQAResource():
+    def __init__(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+        self.dictionary = {}
+
+        self.dir_name = DIR_NAME
+        self.remote_url = 'https://' + GITHUB_AUTH_TOKEN + '@' + REMOTE_URL
+
+    def process_dqa(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+
+        dir_name = './dqa_repo/SecondaryReports/'
+        hospitals = filter(
+            lambda f: (f != 'Ranking'
+                       and os.path.isdir(os.path.join(dir_name, f))),
+            os.listdir(dir_name))
+
+        self.site_totals = {}
+
+        for hospital in hospitals:
+            hosp_path = os.path.join(dir_name, hospital)
+
+            # for each site, we only need data from the latest report for each
+            # data model version, so go through the dqa reports in reverse
+            # order and ignore older reports for each data model version.
+            for etl_version in sorted(os.listdir(hosp_path), reverse=True):
+                etl_path = os.path.join(hosp_path, etl_version)
+
+                if etl_version.startswith('ETL') and os.path.isdir(etl_path):
+                    file_names = list(filter(lambda x: x.endswith('.csv'),
+                                             os.listdir(etl_path)))
+
+                    version = Parser.get_version(
+                        codecs.open(os.path.join(etl_path, file_names[0]), 'r',
+                                    encoding='latin1'))
+
+                    if hospital in self.site_totals.setdefault(version, {}):
+                        continue
+                    else:
+                        self.site_totals[version][hospital] = {}
+
+                    self.table_totals.setdefault(version, {})
+                    self.field_totals.setdefault(version, {})
+
+                    for file_name in file_names:
+                        table = file_name.split('.csv')[0]
+                        buff = codecs.open(
+                            os.path.join(etl_path, file_name), 'r',
+                            encoding='latin1')
+                        p = Parser(buff, hospital, table,
+                                   self.field_totals[version],
+                                   self.table_totals[version],
+                                   self.site_totals[version][hospital])
+                        p.parse()
+
+    def process_dictionary(self):
+        path = './dqa_repo/Dictionary/DCC_DQA_Dictionary.csv'
+        buff = codecs.open(path, 'r', encoding='latin1')
```

Comment:
I started off assuming utf-8 but the parser was failing on some of the files because of unsupported characters (they were latin-1). I don't remember which files and how many -- maybe some of the older ones? I will check. But if the newly generated files are now all utf-8, the old latin-1 ones are still stuck in there, and I need to deal with them as well.